include/ck/tensor_operation/gpu/device/impl/device_grouped_conv_bwd_weight_xdl_cshuffle.hpp

```diff
@@ -273,72 +315,95 @@ struct DeviceGroupedConvBwdWeight_Xdl_CShuffle
             batch);
     }
 
+    template <typename SeqType>
+    [[nodiscard, gnu::always_inline]] inline constexpr static auto
+    ShuffleSequenceAndTransformFrom4DTo3D() noexcept(noexcept(SeqType{}.Size() == 4))
+    {
+        // Remove first element and,
+        // Convert 4d->3d sequence.
+        constexpr auto _I0  = SeqType{}.At(I1);
+        constexpr auto _I1  = SeqType{}.At(I2);
+        constexpr auto _I2  = SeqType{}.At(I0);
+        constexpr auto _Seq = Sequence<_I0, _I1, _I2>();
+        return _Seq;
+    }
+
+    template <typename SeqType>
+    [[nodiscard, gnu::always_inline]] inline constexpr static auto
```

Comment:
Why nodiscard is needed?

Reply:
It tells us, attribute to a function return value to indicate that the caller must use the result.

Reply:
I want to avoid situation when a function returns value and the caller not use it somehow.